nikola/plugin_categories.py

```diff
@@ -202,10 +201,14 @@ class TemplateSystem(BasePlugin):
 
     name = "dummy_templates"
 
-    def set_directories(self, directories: 'typing.List[str]', cache_folder: str):
+    def set_directories(self, directories: List[str], cache_folder: str) -> None:
         """Set the list of folders where templates are located and cache."""
         raise NotImplementedError()
 
+    def user_engine_factory(self, factory: Callable[..., Any]) -> None:
```

Comment:
Consider a different name that includes a verb. Perhaps `set_user_engine_factory`?